src/frontend/src/pages/GanttPage/GanttChartComponents/GanttTaskBar/GanttTaskBarEdit.tsx

```diff
@@ -85,169 +94,216 @@ const GanttTaskBarEdit = ({
+              },
+              stage: workPackageInfo.stage,
+              projectName: task.name,
+              status: WbsElementStatus.Inactive,
+              orderInProject: newWorkPackageNumber,
+              duration: 1,
+              immediatelyBlocking: [],
+              descriptionBullets: [],
+              links: [],
+              wbsElementId: '-1',
+              dateCreated: new Date(),
+              lead: undefined,
+              manager: undefined,
+              changes: [],
+              materials: [],
+              assemblies: []
+            };
+            addWorkPackage(workPackage);
+
+            createChange({
+              id,
+              type: 'create-work-package',
+              element: workPackage
+            });
+          }}
+        />
+        <Box
+          sx={{
+            width: '100%',
+            display: 'grid',
+            gap: GANTT_CHART_GAP_SIZE,
+            gridTemplateRows: `repeat(1, minmax(0, 1fr))`,
+            gridTemplateColumns: `repeat(${days.length}, minmax(auto, 1fr))`,
+            position: 'absolute',
+            top: 0,
+            left: 0
+          }}
+        >
+          {/* Drop areas */}
+          {showDropPoints &&
+            days.map((day, index) => (
+              <Box
+                key={index}
+                onDragOver={onDragOver}
+                onDrop={() => onDrop(day)}
+                sx={{
+                  borderRadius: '0.25rem',
+                  height: '2rem',
+                  minWidth: GANTT_CHART_CELL_SIZE,
+                  maxWidth: GANTT_CHART_CELL_SIZE,
+                  backgroundColor: `color-mix(in srgb, ${theme.palette.background.default}, transparent 75%);`
+                }}
+              />
+            ))}
+        </Box>
+        <Box
+          sx={{
+            display: 'grid',
+            gap: GANTT_CHART_GAP_SIZE,
+            gridTemplateRows: `repeat(1, minmax(0, 1fr))`,
+            gridTemplateColumns: `repeat(${days.length}, minmax(${GANTT_CHART_CELL_SIZE}, 1fr))`,
+            width: '100%'
           }}
         >
-          <Box
-            sx={{
-              height: '100%',
-              width: '100%',
-              display: 'flex',
-              justifyContent: 'space-between',
-              alignItems: 'center',
-              overflow: 'visible'
+          <div
+            ref={measureRef}
+            style={{
+              gridColumnStart: getStartCol(task.start),
+              gridColumnEnd: getEndCol(task.end),
+              height: '2rem',
+              width: isProject ? 'unset' : width === 0 ? `unset` : `${width}px`,
+              border: `1px solid ${isResizing ? theme.palette.text.primary : theme.palette.divider}`,
+              borderRadius: '0.25rem',
+              backgroundColor: task.styles ? task.styles.backgroundColor : theme.palette.background.paper,
+              cursor: isProject ? 'default' : 'move'
```

Comment:
I  feel like some of this styling stuff is probably duplicate between these files (here and in a bunch of places), abstract it into a reusable object/ helper method that generates the style object if u can

Reply:
yeah i didnt wanna touch the styling yet, next steps is definitely the abstraction in the task bars rn their pretty awful